packages/round-manager/src/features/round/ApplicationsToApproveReject.tsx

```diff
@@ -0,0 +1,323 @@
+  const filteredApplications = (applications || []).filter((a) =>
+    isDirectRound
+      ? a.inReview
+      : a.status == ApplicationStatus.PENDING.toString()
+  );
+
+  const [bulkSelect, setBulkSelect] = useState(false);
+  const [openModal, setOpenModal] = useState(false);
+  const [openProgressModal, setOpenProgressModal] = useState(false);
+  const [openErrorModal, setOpenErrorModal] = useState(false);
+  const [selected, setSelected] = useState<GrantApplication[]>([]);
+
+  const {
+    bulkUpdateGrantApplications,
+    contractUpdatingStatus,
+    indexingStatus,
+  } = useBulkUpdateGrantApplications();
+  const isBulkUpdateLoading =
+    contractUpdatingStatus == ProgressStatus.IN_PROGRESS ||
+    indexingStatus == ProgressStatus.IN_PROGRESS;
+
+  const progressSteps: ProgressStep[] = [
+    {
+      name: "Updating",
+      description: `Updating application statuses on the contract.`,
+      status: contractUpdatingStatus,
+    },
+    {
+      name: "Indexing",
+      description: "The subgraph is indexing the data.",
+      status: indexingStatus,
+    },
+    {
+      name: "Redirecting",
+      description: "Just another moment while we finish things up.",
+      status:
+        indexingStatus === ProgressStatus.IS_SUCCESS
+          ? ProgressStatus.IN_PROGRESS
+          : ProgressStatus.NOT_STARTED,
+    },
+  ];
+
+  useEffect(() => {
+    if (!isLoading || !bulkSelect) {
+      setSelected(
+        (filteredApplications || []).map((application) => {
+          return {
+            id: application.id,
+            round: application.round,
+            recipient: application.recipient,
+            projectsMetaPtr: application.projectsMetaPtr,
+            status: application.status,
+            applicationIndex: application.applicationIndex,
+            createdAt: application.createdAt,
+          };
+        })
+      );
+    }
+  }, [applications, isLoading, bulkSelect]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (contractUpdatingStatus === ProgressStatus.IS_ERROR) {
+      setTimeout(() => {
+        setOpenErrorModal(true);
+        setOpenProgressModal(false);
+      }, errorModalDelayMs);
+    }
+
+    if (indexingStatus === ProgressStatus.IS_SUCCESS) {
+      window.location.reload();
+    }
+  }, [contractUpdatingStatus, indexingStatus]);
+
+  const handleDone = () => {
+    window.location.reload();
+  };
+
+  const toggleSelection = (id: string, status: ProjectStatus) => {
+    const newState = selected?.map((grantApp: GrantApplication) => {
+      if (grantApp.id === id) {
+        const newStatus = grantApp.status === status ? "PENDING" : status;
+        return { ...grantApp, status: newStatus };
+      }
+
+      return grantApp;
+    });
+
+    setSelected(newState);
+  };
+
+  const checkSelectionStatus = (id: string) => {
+    return selected?.find((grantApp: GrantApplication) => grantApp.id === id)
+      ?.status;
+  };
```

Comment:
This appears to be returning the selection status rather than checking it. `getSelectionStatus` would be more in line with what it does.